File: sejal_mumbai/sejal_mumbai/report/items_report/items_report.js
frappe.query_reports["Items Report"] = {
    "filters": [
        {
            "fieldname": "name",
            "label": __("Name"),
            "fieldtype": "Link",
            "options": "Item",
        },
        {
            "fieldname": "custom_karigar",
            "label": __("Karigar"),
            "fieldtype": "Link",
            "options": "Karigar",
        },
        {
            "fieldname": "custom_warehouse",
            "label": __("Location"),
            "fieldtype": "Link",
            "options": "Warehouse",
        },
        {
            "fieldname": "limit_page_length",
            "label": __("Page Length"),
            "fieldtype": "Select",
            "options": ["10", "20", "50", "100", "500", "ALL"], // Corrected 500 option
            "default": "10",
        },
        {
            "fieldname": "limit_start",
            "label": __("Start"),
            "fieldtype": "Int",
            "default": 0,
        },
    ],
}

var selectedValue = 10;
function execute(value) {
   
    
    frappe.query_report.filters[2].set_value(value); // Update page length filter value
    frappe.query_report.refresh(); // Refresh report with updated filters
}

// Create pagination buttons and append them to the DOM
var div = document.createElement('div');
div.className = 'list-paging-area level';
div.innerHTML = `
    <div class="level-left">
        <div class="btn-group">
            <button type="button" class="btn btn-default btn-sm btn-paging ${selectedValue === 10 ? 'btn-info' : ''}" data-value="10">
                10
            </button>
            <button type="button" class="btn btn-default btn-sm btn-paging ${selectedValue === 20 ? 'btn-info' : ''}" data-value="20">
                20
            </button>
            <button type="button" class="btn btn-default btn-sm btn-paging ${selectedValue === 50 ? 'btn-info' : ''}" data-value="50">
                50
            </button>
            <button type="button" class="btn btn-default btn-sm btn-paging ${selectedValue === 100 ? 'btn-info' : ''}" data-value="100">
                100
            </button>
			<button type="button" class="btn btn-default btn-sm btn-paging ${selectedValue === 500 ? 'btn-info' : ''}" data-value="500">
			    500 <!-- Corrected data-value to 500 -->
			</button>
            <button type="button" class="btn btn-default btn-sm btn-paging ${selectedValue === 'ALL' ? 'btn-info' : ''}" data-value="ALL">
                ALL
            </button>
        </div>
    </div>
 
`;
document.querySelector('.layout-main-section.frappe-card').appendChild(div);

// Add event listeners to pagination buttons
document.querySelectorAll('.btn-paging').forEach(button => {
    button.addEventListener('click', function() {
        var value = this.getAttribute('data-value');
        if (selectedValue === value) {
            return;
        } else {
            // Remove 'btn-info' class from all buttons
            document.querySelectorAll('.btn-paging').forEach(btn => {
                btn.classList.remove('btn-info');
            });
            // Add 'btn-info' class to the clicked button
            this.classList.add('btn-info');
            selectedValue = value;
            // Call execute function with the selected value
            execute(value);
        }
    });
});

File: sejal_mumbai/sejal_mumbai/report/items_report/items_report.py
import frappe
from frappe import _

def execute(filters=None):
    limit_start = filters.get("limit_start", 0)
    limit_page_length = filters.get("limit_page_length", 10)
    if limit_page_length == "ALL":
        limit_clause = ""
    else:
        limit_clause = f"LIMIT {limit_start}, {limit_page_length}"
    conditions = get_conditions(filters)
    columns = [
        {
            "label": "Item Code",
            "fieldname": "name",
            "fieldtype": "Link",
            "options": "Item",
            "width": 200,
        },
        {
            "label": "Custom Gross Wt",
            "fieldname": "custom_gross_wt",
            "fieldtype": "Data",
            "width": 200,
        },
        {
            "label": "Custom Net Wt",
            "fieldname": "custom_net_wt",
            "fieldtype": "Data",
            "width": 200,
        },
        {
            "label": "Karigar",
            "fieldname": "custom_karigar",
            "fieldtype": "Link",
            "options": "karigar",
            "width": 200,
        },
        {
            "label": "Location",
            "fieldname": "custom_warehouse",
            "fieldtype": "Link",
            "options": "Warehouse",
            "width": 200,
        },
    ]

    data = frappe.db.sql(
        f"""
        SELECT
            item.name,
            item.custom_gross_wt,
            item.custom_net_wt,
            item.custom_karigar,
            item.custom_warehouse as custom_warehouse
        FROM `tabItem` as item
        {conditions}
        {limit_clause}
        """,
        as_dict=True,
    )
    return columns, data

def get_conditions(filters):
    conditions = []
    if filters and filters.get("name") and filters.get("name") != "":
        conditions.append(f'item.name = "{filters.get("name")}"')
    if filters and filters.get("custom_karigar") and filters.get("custom_karigar") != "":
        conditions.append(f'item.custom_karigar = "{filters.get("custom_karigar")}"')
    if filters and filters.get("custom_warehouse") and filters.get("custom_warehouse") != "":
        conditions.append(f'item.custom_warehouse = "{filters.get("custom_warehouse")}"')
           

    if conditions:
        return "WHERE " + " AND ".join(conditions)
    else:
        return ""
